cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -161,3 +161,435 @@ fn read_output_to_channel(
+                Output::Exit(status) => match expect_success {
+                    true => assert!(status.success()),
+                    false => assert!(!status.success()),
+                },
+                // Copy output to debug failing tests.
+                Output::Stdout(line) => println!("stdout: {line}"),
+                Output::Stderr(line) => println!("stderr: {line}"),
+            }
+        }
+    }
+
+    // Asserts that `file_name` exists with `content` within the build result
+    // for `package_name`.
+    fn assert_build_file(
+        parent: &Path,
+        package_name: &String,
+        file_name: &String,
+        content: &String,
+    ) {
+        let dir = result_dir(parent, package_name);
+        assert!(dir.is_symlink());
+        assert!(dir.read_link().unwrap().starts_with("/nix/store/"));
+
+        let file = dir.join(file_name);
+        assert!(file.is_file());
+        assert_eq!(fs::read_to_string(file).unwrap(), **content);
+    }
+
+    #[test]
+    fn build_returns_failure_when_package_not_defined() {
+        let package_name = String::from("foo");
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, "version = 1");
+
+        assert_build_status(&flox, &mut env, &package_name, false);
+    }
+
+    #[test]
+    fn build_command_generates_file() {
+        let package_name = String::from("foo");
+        let file_name = String::from("bar");
+        let file_content = String::from("some content");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            command = """
+                mkdir $out
+                echo -n {file_content} > $out/{file_name}
+            """
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+        let env_path = env.parent_path().unwrap();
+
+        assert_build_status(&flox, &mut env, &package_name, true);
+        assert_build_file(&env_path, &package_name, &file_name, &file_content);
+    }
+
+    #[test]
+    #[ignore] // TODO: `files` isn't currently passed to or parsed by `flox-build.mk`.
+    fn build_includes_files() {
+        let package_name = String::from("foo");
+        let file_name = String::from("bar");
+        let file_content = String::from("some content");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            files = ["{file_name}"]
+            command = "mkdir $out"
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+        let env_path = env.parent_path().unwrap();
+
+        fs::write(env_path.join(&file_name), &file_content).unwrap();
+
+        assert_build_status(&flox, &mut env, &package_name, true);
+        assert_build_file(&env_path, &package_name, &file_name, &file_content);
+    }
```

Comment:
There are oen questions regarding design for this, let alone an issue or implementation.
Do we even need to include the test if the tested functionality does not exist?

Reply:
Follow up in person